src/modules/uORB/uORBDeviceNode.cpp

```diff
@@ -44,12 +44,54 @@
 
 static uORB::SubscriptionInterval *filp_to_subscription(cdev::file_t *filp) { return static_cast<uORB::SubscriptionInterval *>(filp->f_priv); }
 
+// Determine the data range
+template <typename T>
+static inline bool is_in_range(const T &left, const T &value, const T &right)
+{
+	if (right > left) {
+		return (left <= value) && (value <= right);
+
+	} else {  // Maybe the data overflowed and a wraparound occurred
+		return (left <= value) || (value <= right);
+	}
+}
+
+// round up to nearest power of two
+// Such as 0 => 1, 1 => 1, 2 => 2 ,3 => 4, 10 => 16, 60 => 64, 65...255 => 128
+// Note: When the input value > 128, the output is always 128
+static inline uint8_t roundup_pow_of_two_8(uint8_t n)
+{
+	if (n == 0) {
```

Comment:
It might just be me, but since this code is only called once or twice per publication I'd prefer something more readable than the bitwise methods.

``` C++
int next = pow(2, ceil(log(x)/log(2)));
return math::constrain(next, 1, UINT8_MAX);
```

Reply:
I simplified the implementation, it may be a little better now.
```C++
	if (n == 0) {
		return 1;
	}

	// Avoid is already a power of 2
	uint32_t value = n - 1;

	// Fill 1
	value |= value >> 1U;
	value |= value >> 2U;
	value |= value >> 4U;

	// Unable to round-up, take the value of round-down
	if (value == UINT8_MAX) {
		value >>= 1U;
	}

	return value + 1;
```

Reply:
math::constrain(next, 1, UINT8_MAX);
We cannot use UINT8_MAX, which is 255, and the maximum can only be 128.
I think we should change the internal _queue_size to uint16, so that it can be 256.